core/src/main/java/nl/nn/adapterframework/validation/XercesXmlValidator.java

```diff
@@ -257,23 +265,31 @@ public XercesValidationContext createValidationContext(IPipeLineSession session,
 		result.init(schemasProvider, preparseResult.getSchemasId(), preparseResult.getNamespaceSet(), rootValidations, invalidRootNamespaces, ignoreUnknownNamespaces);
 		return result;
 	}
-	
 
-	public ValidatorHandler getValidatorHandler(IPipeLineSession session, ValidationContext context) throws ConfigurationException, PipeRunException {
+	public ValidatorHandlerImpl getValidatorHandler(IPipeLineSession session, ValidationContext context) throws ConfigurationException {
+		// User Impl version from xerces, because it has extra validate() function.
+		ValidatorHandlerImpl validatorHandler;
 
-		ValidatorHandler validatorHandler=null;
 		try {
-			XMLSchemaFactory schemaFactory = new XMLSchemaFactory();
-			javax.xml.validation.Schema schemaObject=schemaFactory.newSchema(((XercesValidationContext)context).getGrammarPool());
-			validatorHandler=schemaObject.newValidatorHandler();
+			javax.xml.validation.Schema schemaObject;
+			if (xsdVersion == 1.0) {
+				XMLSchemaFactory schemaFactory = new XMLSchemaFactory();
+				schemaObject = schemaFactory.newSchema(((XercesValidationContext) context).getGrammarPool());
+			} else {
+				XMLSchema11Factory schemaFactory = new XMLSchema11Factory();
+				schemaObject = schemaFactory.newSchema(((XercesValidationContext) context).getGrammarPool());
+			}
+
+			validatorHandler = (ValidatorHandlerImpl) schemaObject.newValidatorHandler();
 		} catch (SAXException e) {
 			throw new ConfigurationException(logPrefix + "Cannot create schema", e);
 		}
 		try {
-			//validatorHandler.setFeature(NAMESPACES_FEATURE_ID, true);
 			validatorHandler.setFeature(VALIDATION_FEATURE_ID, true);
 			validatorHandler.setFeature(SCHEMA_VALIDATION_FEATURE_ID, true);
 			validatorHandler.setFeature(SCHEMA_FULL_CHECKING_FEATURE_ID, isFullSchemaChecking());
+
+			validatorHandler.setContentHandler(context.getContentHandler());
```

Comment:
Why is this necessary now, and wasn't it before? Was it a bug?

Reply:
It was done before as well (see: AbstractXmlValidator.java:213) It lets us check known namespaces. Since I overrode validate function, I needed to re-set it.